dawn/src/dawn/CodeGen/Cuda-ico/ASTStencilBody.cpp

```diff
@@ -152,9 +124,19 @@ void ASTStencilBody::visit(const std::shared_ptr<iir::FunCallExpr>& expr) {
 }
 
 void ASTStencilBody::visit(const std::shared_ptr<iir::ReductionOverNeighborExpr>& expr) {
+  DAWN_ASSERT_MSG(!parentIsReduction_,
+                  "Nested Reductions not yet supported for CUDA code generation");
+
   std::string lhs_name = "lhs_" + std::to_string(expr->getID());
+
+  if(!firstPass_) {
+    ss_ << " " << lhs_name << " ";
+    return;
+  }
+
+  parentIsReduction_ = true;
+
   std::string weights_name = "weights_" + std::to_string(expr->getID());
-  ss_ << "{";
```

Comment:
You removed the opening braces but not the closing braces.

Reply:
> You removed the opening braces but not the closing braces.

I did, the closing bracket for this is on line 105:
<< "lhs_" + std::to_string(reductionFinder.reduceOverNeighborExpr()->getID()) << ";}\n";